### ISSUE TITLE
the plugin make the obsidian UI unable to get the correct locale setting 

### ISSUE THREAD

**Issue by qifei9** [zsviczian/obsidian-excalidraw-plugin#290]
**Describe the bug**
I set the obsidian language to Chinese Simplified, and thus the obsidian UI (i.e. where you change settings of obsidian, install plugin, etc) displays settings in Chinese characters.

After install and enable this plugin, the characters looks weird. It seems the obsidian UI is displaying Japanese characters instead of Chinese characters.

Note, for a CJK font (CJK stands for Chinese, Japanese and Korean), such as "Noto sans CJK", the characters of the 3 languages are all in the same font, and the locale setting decides which characters to display. If the locale is not set correctly, Japanese characters are displayed, since in locale code "jp" is in front of "zh" and "ko" alphabetically.

Thus I guess this issue is because of the obsidian UI cannot get the correct locale setting after enable the plugin

**To Reproduce**
1. set the obsidian language to Chinese Simplified
2. using a CJK font such as Noto Sans CJK (from system setting?)
3. install and enable the plugin
4. See the obsidian UI

**Desktop (please complete the following information):**
 - OS: archlinux updated to date
 - plugin version 1.4.17
 - obsidian Version 0.12.19

**Additional context**
Not sure whether my guess is correct

**Comment by zsviczian** [comment]
This is strange. I am not aware of doing anything to characters. 
Have you tried to uninstall Excalidraw? Did the characters in Obsidian return to normal?
Or are you just talking about characters in Excalidraw settings?

**Comment by qifei9** [comment]
> This is strange. I am not aware of doing anything to characters. Have you tried to uninstall Excalidraw? Did the characters in Obsidian return to normal? Or are you just talking about characters in Excalidraw settings?

1. No, not the characters in Excalidraw settings or in the draw. It's about the characters in the obsidian UI, like the place where you enable/disable core plugins, change themes, etc.
2. When I disable or uninstall the Excalidraw plugin and restart obsidian, everything return to normal.

I explored a little bit more. The problems only appears after every time I open a `.excalidraw` file. If I start obsidian with the Excalidraw plugin installed and enabled, and `.excalidraw` files in the vault, but never open a `.excalidraw`, everything is normal. However, once I open a `.excalidraw` file in obsidian, the problem occurs.

**Comment by zsviczian** [comment]
must be some stylesheet inside the Excalidraw component.
Are you using the legacy .excalidraw format or the markdown excalidraw files?
which theme are you using?

**Comment by qifei9** [comment]
I use the default theme.

>  Are you using the legacy .excalidraw format or the markdown excalidraw files? 

Not sure what do you mean, but I didn't change any setting of the plugin. I create a new draw using the button in the left sidebar, and the file is `Excalidraw/*.excalidraw`, and then I draw in the file.

**Comment by zsviczian** [comment]
In Excalidraw settings do you have these turned off as shown on the image? (BTW I don't think this has anything to do with the error, just trying to explore every possible option to find what could be wrong...)

![image](https://user-images.githubusercontent.com/14358394/144706671-e8936ae5-8e6f-4687-a618-0e338a0a7093.png)

**Comment by qifei9** [comment]
Yes, these 3 settings are all at "off", i.e. my settings are identical to the image

**Comment by zsviczian** [comment]
I tried to reproduce the error but I could not notice any difference.

What other plugin do you have installed?
Can you disable all other community plugins, only leave Excalidraw, restart Obsidian and see if the issue is still there?

Which language are you using? This?
![image](https://user-images.githubusercontent.com/14358394/144706859-a0788dc6-0192-41d1-b6c0-0b078a36216d.png)

**Comment by zsviczian** [comment]
One other request: can you open developer console CTRL+SHIFT+i
and execute the following command before and after starting Excalidraw?
On my side, I don't see a change to local settings as a result of starting Excalidraw. I wonder if there is a change on your side.
![image](https://user-images.githubusercontent.com/14358394/144707024-c76ca585-a3cb-4101-a398-835c99afc76f.png)

**Comment by qifei9** [comment]
1. The language is correct as the arrow pointed to.
2. I tested by creating a new vault, keep everything as default and only install Excalidraw, and the problem still occurs.
3. To produce the error: create a new vault, install and enable Excalidraw, create and open a draw, draw something for a while (like 1-2 min), then the problem occurs and part of the characters in the obsidian UI change, then DO NOT close the `.excalidraw` file, and directly close the obsidian, and then open obsidian again, the obsidian will automate open the `excalidraw`, and now most of the characters in the obsidian UI change.
4. To back to normal, in the obsidian, close the `.excalidraw` file, and open another normal MD file, and then close the obsidian, then open obsidian again, the obsidian will automate open the MD file, and now everything is normal.
5. in the developer console, the locale never change, it's always correct ('zh-ch')

I attached 2 figures showing the problems, and marked a obverse change
normal:
![normal](https://user-images.githubusercontent.com/856187/144707477-7789e162-d119-4e6c-b390-eb3259885875.png)

problem:
![problem_occur](https://user-images.githubusercontent.com/856187/144707484-0a8487a5-d31b-418d-9d08-eafca4a07b01.png)

**Comment by zsviczian** [comment]
Using the developer console can you check if the text in HTML is correct or not, and filter on font style settings and send me the final screenshot.

https://user-images.githubusercontent.com/14358394/144708866-3c8818a9-d3a8-4e71-983d-ddcefd78bf2a.mp4

**Comment by qifei9** [comment]
Things are really weird, but the problem seems resolved. 

1. my font style settings seen in the developer console are identical to yours.
2. the developer console always show the incorrect characters, like in the code
```css
<div class="setting-item-name">半透明效果</div>
```
3. then I found the incorrect characters are actually "KR" rather than "JP" that I thought
4. then I uninstall the Noto Sans CJK font package from the system and reinstall it, the problem disappears.
5. I found this problem is due to that a config file of the font, which is a soft link, was broken. This may due to a previous update of the package, and an uninstall-reinstall points the soft link to the correct place where the actual config file is.

Interestingly, the soft link must have long been broken, this problem never occurs in any other software and in obsidian with ~50 plugins, until I start to draw in the Excalidraw.

Overall, truly sorry to trouble you and many many thanks for your time and this excellent plugin.

normal:
![2021-12-04 20-16-33 的屏幕截图](https://user-images.githubusercontent.com/856187/144714854-b29f73cb-2478-46de-a022-5ddd1dc1337d.png)

problem:
![2021-12-04 20-18-20 的屏幕截图](https://user-images.githubusercontent.com/856187/144714870-97afffd2-d16a-4d70-8b62-e2dd2de31b9f.png)
